client/components/empty-content/index.js

```diff
@@ -0,0 +1,185 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import { startsWith } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { H } from 'layout/section';
+import ImageAsset from 'components/image-asset';
+
+/**
+ * A component to be used when there is no data to show.
+ * It can be used as an opportunity to provide explanation or guidance to help a user progress.
+ */
+class EmptyContent extends Component {
+	renderIllustration() {
+		const { illustrationWidth, illustrationHeight } = this.props;
+		let illustrationSrc = this.props.illustration;
+		let ImageComponent = 'img';
+		if ( startsWith( illustrationSrc, '/' ) ) {
+			ImageComponent = ImageAsset;
+			illustrationSrc = illustrationSrc.substring( 1 );
+		}
+
+		return (
+			<ImageComponent
+				alt=""
+				src={ illustrationSrc }
+				width={ illustrationWidth }
+				height={ illustrationHeight }
+				className="woocommerce-empty-content__illustration"
+			/>
+		);
+	}
+
+	renderPrimaryAction() {
+		const { actionLabel, actionURL, actionCallback } = this.props;
+		if ( actionURL && actionCallback ) {
+			return (
+				<Button
+					className="woocommerce-empty-content__action"
+					isPrimary
+					onClick={ actionCallback }
+					href={ actionURL }
+				>
+					{ actionLabel }
+				</Button>
+			);
+		} else if ( actionURL ) {
+			return (
+				<Button className="woocommerce-empty-content__action" isPrimary href={ actionURL }>
+					{ actionLabel }
+				</Button>
+			);
+		} else if ( actionCallback ) {
+			return (
+				<Button className="woocommerce-empty-content__action" isPrimary onClick={ actionCallback }>
+					{ actionLabel }
+				</Button>
+			);
+		}
+
+		return null;
+	}
+
+	renderSecondaryAction() {
+		const { secondaryActionLabel, secondaryActionURL, secondaryActionCallback } = this.props;
+		if ( secondaryActionURL && secondaryActionCallback ) {
+			return (
+				<Button
+					className="woocommerce-empty-content__secondary-action"
+					onClick={ secondaryActionCallback }
+					href={ secondaryActionURL }
+				>
+					{ secondaryActionLabel }
+				</Button>
+			);
+		} else if ( secondaryActionURL ) {
+			return (
+				<Button className="woocommerce-empty-content__secondary-action" href={ secondaryActionURL }>
+					{ secondaryActionLabel }
+				</Button>
+			);
+		} else if ( secondaryActionCallback ) {
+			return (
+				<Button
+					className="woocommerce-empty-content__secondary-action"
+					onClick={ secondaryActionCallback }
+				>
+					{ secondaryActionLabel }
+				</Button>
+			);
+		}
+
+		return null;
+	}
+
+	renderActions() {
+		const { actionLabel, secondaryActionLabel } = this.props;
+
+		if ( ! actionLabel && ! secondaryActionLabel ) {
+			return null;
```

Comment:
If we make these required proptypes, it'll be a good reminder for devs

Reply:
I made the primary action required. I think we can leave the secondary one as optional.